servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -5559,6 +5596,10 @@ public void deleteServiceIdentity(ResourceContext ctx, String domainName,
 
         verifyAuthorizedServiceOperation(((RsrcCtxWrapper) ctx).principal().getAuthorizedService(), caller);
 
+        // make sure no domain is dependent on the service
+
+        verifyNoDomainDependenciesOnService(domainName + "." + serviceName, caller);
```

Comment:
Our original design only mentioned preventing domain deletion that has dependencies. I've added the other direction as well - prevent deletion of the service provider if it has domain dependencies.